classes/user/DAO.inc.php

```diff
@@ -302,4 +303,26 @@ public function changeSitePrimaryLocale($oldLocale, $newLocale)
             }
         }
     }
+
+    /**
+     * Delete unvalidated expired users
+     * 
+     * @param object<Carbon\Carbon> $dateTillValid      The dateTime till before which user will consider expired
+     * @param array                 $excludableUsersId  The users id to exclude form delete operation
+     * 
+     * @return int The number rows affected by DB operation
+     */
+    public function deleteUnvalidatedExpiredUsers(Carbon $dateTillValid, array $excludableUsersId = []) {
+
+        $users = DB::table('users')
+            ->whereNull('date_validated')
+            ->whereNull('date_last_login')
+            ->where('date_registered', '<', $dateTillValid)
+            ->when( !empty($excludableUsersId), fn($query) => $query->whereNotIn('id', $excludableUsersId) )
```

Comment:
Is there a use case for this parameter? I don't see it used in the PR.

Reply:
@asmecher  Right now none but maybe later if we need to deleted specific filtered users . That way this method can be reused in future while the unvalidated removal logic encapsulated in a single method without duplication . You can say this is a bit over optimisation right now and I agree with it but feels to me up to this level over optimisation is acceptable . However I am open of any suggestions .

Reply:
I'm OK to leave this in place, but in general I'd recommend not adding methods/options unless we have a use case for them. They're easy to add later but as this isn't used/tested it's just dead code until then... and we have a lot of dead code :)

Reply:
ok, got it . I will refrain from doing it in future .